operator/src/main/java/oracle/kubernetes/operator/steps/ManagedServersUpStep.java

```diff
@@ -118,23 +118,30 @@ public NextAction apply(Packet packet) {
   private void addServersToFactory(@Nonnull ServersUpStepFactory factory, @Nonnull WlsDomainConfig wlsDomainConfig) {
     Set<String> clusteredServers = new HashSet<>();
 
+    List<ServerConfig> pendingServers = new ArrayList<>();
     wlsDomainConfig.getClusterConfigs().values()
-        .forEach(wlsClusterConfig -> addClusteredServersToFactory(factory, clusteredServers, wlsClusterConfig));
+        .forEach(wlsClusterConfig -> addClusteredServersToFactory(
+            factory, clusteredServers, wlsClusterConfig, pendingServers));
 
     wlsDomainConfig.getServerConfigs().values().stream()
         .filter(wlsServerConfig -> !clusteredServers.contains(wlsServerConfig.getName()))
-        .forEach(wlsServerConfig -> factory.addServerIfNeeded(wlsServerConfig, null));
+        .forEach(wlsServerConfig -> factory.addServerIfAlways(wlsServerConfig, null, pendingServers));
+
+    for (ServerConfig serverConfig : pendingServers) {
+      factory.addServerIfNeeded(serverConfig.wlsServerConfig, serverConfig.wlsClusterConfig);
```

Comment:
Would this approach affect the guaranteed 'lexi numeric' order in which we start or shutdown servers?  Or which servers are reported in status?   (For example, when we're shutting down a cluster's servers one-at-a-time, the goal is to shutdown only the 'highest' server first, then the second highest, and so on.)

Reply:
The approach considers this. The pending list is in the original order. We have unit test cases covering this too.

Reply:
But the final list does not maintain the original order. For example, if server3 is always and replicas is 2, server1 and server3 will be started, and server3 will be started before server1.

Reply:
In the sample example, if cluster later scales down, server1 will be taken down.

Reply:
The only behavior difference is the startup order among servers that need to be started up in the same round of make right check; servers with ALWAYS policy will be started before the servers with If-needed policy.

Reply:
We want the overall startup and shutdown order to be very intuitive, predictable, and the exact reverse of each-other.

Question 1: I think it'd be better if server1 always first in a cluster, then server2, and so on, if they aren't configured to start concurrently -- regardless of whether any of the servers are marked 'always'.  So, based on your analysis of the cluster use case ^^^ where server3 (marked always) starts before server1 (marked if_needed), it sounds like this pull may need to be refined?

Question 2:  As for shutdown, if replica count is 3, and server1 & 3 are 'if needed' while server2 is always, then reducing replica count to 1 (or 0) should always shutdown server3 first and then shutdown server1 second.  (Side note: when setting the entire cluster to NEVER the entire cluster is expected to shutdown concurrently, so no worries there.)  Based on your analysis, it sounds like this will be honored?

Reply:
I understand why you started the ALWAYS servers first, but I agree that it's more important that servers be started (and stopped) in a predictable and consistent order. We've had a few customers ask about startup ordering, so I could see us doing something like that in a future release. I think that means we need to keep the order very, very simple now so that customers would understand what they are configuring later.

Reply:
"Yes" to both questions. Good catch on sequential startup order. I think we need to resort the final list.

Reply:
Fixed. 
Kicked off a new Jenkins job https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/2893/.